geom/geom/src/TGeoManager.cxx

```diff
@@ -749,6 +749,27 @@ Int_t TGeoManager::AddRegion(TGeoRegion *region)
   return size;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Add a user-defined property
+
+void TGeoManager::AddProperty(TString property, Double_t value)
+{
+   auto pos = fProperties.insert(ConstPropMap_t::value_type(property, value));
+   if (!pos.second)
+      Warning("AddProperty", "Property \"%s\" already exists with value %g", property.Data(), (pos.first)->second);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Get a user-defined property
+
+Bool_t TGeoManager::GetProperty(const char *property, Double_t &value) const
```

Comment:
This is maybe better than the TString in the interface. Now, often we see that the error code is optional and passed by reference and the return value returned. Maybe this pattern can be adopted? (e.g. `TInterpreter::ProcessLine`)

Reply:
Good point, now returning the requested value for the property and filling a bool as error code if provided. Thanks for the useful comments!